This bug was reported as <https://github.com/apache/arrow/issues/44754> and is fixed on the main branch

--- a/cpp/src/arrow/util/io_util.cc
+++ b/cpp/src/arrow/util/io_util.cc
@@ -118,7 +118,7 @@
 #endif
 
 #ifdef _WIN32
-#  include <Windows.h>
+#  include <windows.h>
 #else
 #  include <dlfcn.h>
 #endif
